checker/tests/index/CustomContractWithArgs.java

```diff
@@ -0,0 +1,170 @@
+import org.checkerframework.checker.index.qual.LTLengthOf;
+import org.checkerframework.common.value.qual.MinLen;
+import org.checkerframework.framework.qual.ConditionalPostconditionAnnotation;
+import org.checkerframework.framework.qual.PostconditionAnnotation;
+import org.checkerframework.framework.qual.PreconditionAnnotation;
+
+public class CustomContractWithArgs {
+    // Postcondition for MinLen
+    @PostconditionAnnotation(
+        qualifier = MinLen.class,
+        sourceArguments = "targetValue",
+        targetArguments = "value"
+    )
+    @interface EnsuresMinLen {
+        public String[] value();
+
+        public int targetValue();
+    }
+
+    // Conditional postcondition for LTLengthOf
+    @ConditionalPostconditionAnnotation(
+        qualifier = LTLengthOf.class,
+        sourceArguments = {"targetValue", "targetOffset"},
+        targetArguments = {"value", "offset"}
+    )
+    @interface EnsuresLTLIf {
+        public boolean result();
+
+        public String[] expression();
+
+        public String[] targetValue();
+
+        public String[] targetOffset();
+    }
+
+    //Precondition for LTLengthOf
+    @PreconditionAnnotation(
+        qualifier = LTLengthOf.class,
+        sourceArguments = {"targetValue", "targetOffset"},
+        targetArguments = {"value", "offset"}
+    )
+    @interface RequiresLTL {
+        public String[] value();
+
+        public String[] targetValue();
+
+        public String[] targetOffset();
+    }
+
+    class Base {
+        @EnsuresMinLen(value = "#1", targetValue = 10)
+        void minLenContract(int[] a) {
+            if (a.length < 10) throw new RuntimeException();
+        }
+
+        @EnsuresMinLen(value = "#1", targetValue = 10)
+        // :: error: (contracts.postcondition.not.satisfied)
+        void minLenWrong(int[] a) {
+            if (a.length < 9) throw new RuntimeException();
+        }
+
+        void minLenUse(int[] b) {
+            minLenContract(b);
+            int @MinLen(10) [] c = b;
+        }
+
+        public int b, y;
+
+        @EnsuresLTLIf(
+            expression = "b",
+            targetValue = {"#1", "#1"},
+            targetOffset = {"#2 + 1", "10"},
+            result = true
+        )
+        boolean ltlPost(int[] a, int c) {
+            if (b < a.length - c - 1 && b < a.length - 10) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        // TODO: this should be an error, because targetOffset refers to non-existent parameter
```

Comment:
I agree that this should be an error. If you add an `@JavaExpression` annotation to the definition of `targetOffset`, does this still fail? So, the definition of `@EnsuresLTLIf` would be:

```java
@interface EnsuresLTLIf {
         public boolean result();
 
         public String[] expression();
 
         @JavaExpression
         public String[] targetValue();
 
         @JavaExpression
         public String[] targetOffset();
     }
```

(see the definition of `LTLengthOf`)

Reply:
Unfortunately, that's how errors that occur when parsing arguments to annotations are reported in general. This is a known issue with `FlowExpressionParseUtil`, so the "errors in the annotations" is actually the expected behavior. I'd add the `@JavaExpression` annotation, since it should be present, even if the behavior isn't exactly what you'd prefer.